Coding/fw/test/TestMain.cpp

```diff
@@ -12,739 +12,807 @@ Main Test entry point
-  TEST_ASSERT_EQUAL_UINT32_MESSAGE(0x00000000, testingCANAdapter.m_outputFrame.m_id, "ID");
-  TEST_ASSERT_EQUAL_UINT8_MESSAGE(0, testingCANAdapter.m_outputFrame.m_dlc, "DLC");
-  TEST_ASSERT_FALSE_MESSAGE(testingCANAdapter.m_outputFrame.m_extended, "Extended");
-  TEST_ASSERT_FALSE_MESSAGE(testingCANAdapter.m_outputFrame.m_rtr, "RTR");
-  TEST_ASSERT_EQUAL_UINT8_MESSAGE(0, testingCANAdapter.m_outputFrame.m_data[0], "DATA0");
-  TEST_ASSERT_EQUAL_UINT8_MESSAGE(0, testingCANAdapter.m_outputFrame.m_data[1], "DATA1");
-  TEST_ASSERT_EQUAL_UINT8_MESSAGE(0, testingCANAdapter.m_outputFrame.m_data[2], "DATA2");
-  TEST_ASSERT_EQUAL_UINT8_MESSAGE(0, testingCANAdapter.m_outputFrame.m_data[3], "DATA3");
-  TEST_ASSERT_EQUAL_UINT8_MESSAGE(0, testingCANAdapter.m_outputFrame.m_data[4], "DATA4");
-  TEST_ASSERT_EQUAL_UINT8_MESSAGE(0, testingCANAdapter.m_outputFrame.m_data[5], "DATA5");
-  TEST_ASSERT_EQUAL_UINT8_MESSAGE(0, testingCANAdapter.m_outputFrame.m_data[6], "DATA6");
-  TEST_ASSERT_EQUAL_UINT8_MESSAGE(0, testingCANAdapter.m_outputFrame.m_data[7], "DATA7");
+    testingCANAdapter.clearOutputFrame();
+    testingCANAdapter.m_currentState = CANInterface::NORMAL;
+    runProtocolExecute(testingSerialAdapter.writeInput("R000000000"));
+    TEST_ASSERT_EQUAL_UINT32_MESSAGE(0x0000000, testingCANAdapter.m_outputFrame.m_id, "ID");
+    TEST_ASSERT_EQUAL_UINT8_MESSAGE(0, testingCANAdapter.m_outputFrame.m_dlc, "DLC");
+    TEST_ASSERT_TRUE_MESSAGE(testingCANAdapter.m_outputFrame.m_extended, "Extended");
+    TEST_ASSERT_TRUE_MESSAGE(testingCANAdapter.m_outputFrame.m_rtr, "RTR");
+    TEST_ASSERT_EQUAL_UINT8_MESSAGE(0, testingCANAdapter.m_outputFrame.m_data[0], "DATA0");
+    TEST_ASSERT_EQUAL_UINT8_MESSAGE(0, testingCANAdapter.m_outputFrame.m_data[1], "DATA1");
+    TEST_ASSERT_EQUAL_UINT8_MESSAGE(0, testingCANAdapter.m_outputFrame.m_data[2], "DATA2");
+    TEST_ASSERT_EQUAL_UINT8_MESSAGE(0, testingCANAdapter.m_outputFrame.m_data[3], "DATA3");
+    TEST_ASSERT_EQUAL_UINT8_MESSAGE(0, testingCANAdapter.m_outputFrame.m_data[4], "DATA4");
+    TEST_ASSERT_EQUAL_UINT8_MESSAGE(0, testingCANAdapter.m_outputFrame.m_data[5], "DATA5");
+    TEST_ASSERT_EQUAL_UINT8_MESSAGE(0, testingCANAdapter.m_outputFrame.m_data[6], "DATA6");
+    TEST_ASSERT_EQUAL_UINT8_MESSAGE(0, testingCANAdapter.m_outputFrame.m_data[7], "DATA7");
+
+    testingCANAdapter.clearOutputFrame();
+    testingCANAdapter.m_currentState = CANInterface::NORMAL;
+    runProtocolExecute(testingSerialAdapter.writeInput("R1FFFFFFF8"));
+    TEST_ASSERT_EQUAL_UINT32_MESSAGE(0x1FFFFFFF, testingCANAdapter.m_outputFrame.m_id, "ID");
+    TEST_ASSERT_EQUAL_UINT8_MESSAGE(8, testingCANAdapter.m_outputFrame.m_dlc, "DLC");
+    TEST_ASSERT_TRUE_MESSAGE(testingCANAdapter.m_outputFrame.m_extended, "Extended");
+    TEST_ASSERT_TRUE_MESSAGE(testingCANAdapter.m_outputFrame.m_rtr, "RTR");
+    TEST_ASSERT_EQUAL_UINT8_MESSAGE(0, testingCANAdapter.m_outputFrame.m_data[0], "DATA0");
```

Comment:
I suggest you write a helper function for the 8 bytes payload compare, sometimg like 

uint8_t expected[8] = {0,0,0,0,0,0,0,0}
checkPayload( frame_m_data, expected);

my Rule of thumb is to extract everything in a helper if it is uses more than 4 lines and repeats more than 2 times